acceptance/tests/pxp-module-puppet/run_puppet_result_failed.rb

```diff
@@ -0,0 +1,70 @@
+require 'pxp-agent/test_helper.rb'
+require 'puppet/acceptance/environment_utils'
+
+test_name 'C93063 - pxp-module-puppet run expect puppet failure' do
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type = test_file_name = File.basename(__FILE__, '.*')
+  tmp_environment   = mk_tmp_environment(app_type)
+  tmp_environment2  = mk_tmp_environment(app_type)
+
+  step 'On master, create a new environment that will result in Failed' do
+    site_manifest = File.join(environmentpath, tmp_environment, 'manifests', 'site.pp')
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  exec{'nonesuchbinary':}
+}
+SITEPP
+    on master, "chmod -R 644 #{File.join(environmentpath, tmp_environment)}"
+  end
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent),
+                         pxp_config_json_using_puppet_certs(master, agent).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  def assert_failed_agent_run(environment, action_result)
+    assert(action_result.has_key?('status'),
+           "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+    assert_equal(environment, action_result['environment'],
+                 "Result of pxp-module-puppet run on #{agent} should run with the "\
+                 "#{environment} environment")
+    assert_equal('unchanged', action_result['status'],
```

Comment:
very strange.  so this definitely does not exit 0 when puppet is run.  but pxp reports unchanged. 
similar with the `error_type` i removed below.

```
[root@fn9m821ucm8kkdi ~]# puppet apply -e "exec{'nonesuchbinary':}"
Notice: Compiled catalog for fn9m821ucm8kkdi.delivery.puppetlabs.net in environment production in 0.07 seconds
Error: Validation of Exec[nonesuchbinary] failed: 'nonesuchbinary' is not qualified and no path was specified. Please qualify the command or specify a path. at line 1
[root@fn9m821ucm8kkdi ~]# echo $?
1
```